spec-examples/src/chacha20_poly1305/chacha20poly1305.rs

```diff
@@ -53,13 +47,7 @@ pub fn decrypt(
     let padded_msg = pad_aad_msg(aad, cipher_text);
     let my_tag = poly(&padded_msg, mac_key);
     if my_tag == tag {
-        match chacha(key, iv, cipher_text) {
```

Comment:
While we're here can you make it so that we always decrypt but only use the result `if my_tag == tag`?